bin/FileSystem.js

```diff
@@ -0,0 +1,24 @@
+module.exports = class FileSystem {
+
+  constructor(driver) {
+    this.driver = driver;
+  }
+
+  async writeFile(data, projectRelativeDirectory, filename, reporter = console.log) {
```

Comment:
consider passing `reporter` in `constructor(...)` if it's not likely to change at runtime (it's usually always the same logging method) as it simplifies the contract with the upper layer